docs/testing/e2e-testing-standards.md

```diff
@@ -0,0 +1,262 @@
+
+- **Facilitates reviews:** Code can be examined more quickly. Since you can quickly determine what the test code is doing, it eases the mental strain of code review.
+
+- **Faster onboarding:** By following name conventions, new hires are onboarded more rapidly and feel at ease enough to contribute to the codebase.
+
+Here are the test standards and guidelines we adhere to when creating Playwright tests at ownCloud.
+
+## Folder Structure:
+- `tests/:`
+
+    - `e2e/`: Main folder containing all E2E test-related files.
+
+        - `features/`: Contains Gherkin feature files.
+
+            - `<specificFeatureFolder>/`: Collection house for "**related"** feature files.
+
+                - `<individualFeatureFile>.feature`: Individual feature files.
+
+        - `pageObjects/`: Contains the Page Object classes.
+
+            - `<specificPageObjectFolder>/`: Collection house for related page objects for each webpage or component
+
+                - `<individualPageObject>.ts`: Page Object for each webpage or component.
+
+        - `steps/`: Holds the step definition files for mapping Gherkin steps to code.
+
+            - `<featureRelatedName>.ts`: Step definitions for each feature.
+
+        - `utils/`: Utility functions and common helpers.
+
+            - `helpers.ts`: Common utility functions (e.g., date formatting, data generation).
+
+        - `config/`: Configuration files for Playwright and other tools.
+
+            - `playwright.config.ts`: Playwright configuration.
+
+        - `hooks/`: Cucumber hooks for setting up and tearing down test environments.
+
+            - `hooks.ts`: Contains `Before`, `After`, and other lifecycle hooks.
+
+        - `reports/`: Generated test reports (e.g., HTML, JSON).
+
+        - `screenshots/`: Captured screenshots during test execution.
+
+        - `videos/`: Recorded videos of test runs.
+
+        - `test-data/`: Static test data files or files for upload.
+
+            - `data.json`: Example JSON files or CSVs with test data.
+
+
+
+## Test Structure - Arrange, Act, Assert:
+
+We can follow the AAA (Arrange, Act, Assert) pattern when structuring the tests. In most cases, the Arrange step can be included in a Before block(hook).
+Consider including comments defining each section to ease readability.
+
+```typescript
+// arrange, create a let propert
+await createProperty()
+
+// act, raise a charge
+// here we can call the method(s) defined in page objects too
+await raiseCharge()
+
+// assert, confirm charge has been raised
+expect(charge).toBe('raised')
+```
+
+## Page Object Model (POM)
+
+Every page should possess one POM file to enhance maintainability and scalability of our tests. It should include all the selectors and functions which belong to corresponding POM.
```

Comment:
```suggestion
Every page should possess one POM file to enhance maintainability and scalability of our tests. It should include all the selectors and functions which are needed to interact with the UI "page".
```

Maybe also explain somewhere what is considered to be a "page". It is not always just a full screen of the UI. It could be some logical part of the UI (dialog...?)

Reply:
I mentioned like this:
"with the UI page or component(s)"